### DISCUSSION TITLE
Error during the first execution of cineast + cottontaildb

### DISCUSSION THREAD

**Discussion by newtonsp** [vitrivr/cineast#248] [Q&A] [answered]
Dear devs,

After a problem in my laptop, I needed to do a fresh vitrivr install. This time I considered the latest stack version and followed instructions available in the GitHub repositories. After installing cottontaildb with apparent success from the source, I did the same with cineast. I was not able to run cottontaildb-examples due to a missing class (ExamplesKT), but one of the attachments suggest that cottontaildb is running as expected. I could also compile cineast without errors.

Unfortunately, when I tried to setup cottontaildb from cineast, I had an error. I attached a print screen from my terminal with cineast error details. 

Sorry if this repository is not the most appropriate one to ask about cineast + cottontaildb. I have a few questions:
1)Do you know if I need to change something in the database or code to fill in the dbo column?
2)Where can I find the sources from the package org.vitrivr.cottontail.client? I did not find them in cineast nor in cottontaildb.
3)Is there an easier way to access cottontaildb data than using GRPC? I remember that Adampro allowed us to use the browser to access the database. For now, the ls command seems to be enough.

My environment has Windows 10 with Oracle JDK 11. I changed a few cineast classes to include the feature module we implemented in the past, but even if I use the original feature modules the error is kept. 

Thank you
sincerely

![cottontaildb](https://user-images.githubusercontent.com/20126691/150570160-d68c738b-1c76-4bea-a05c-b9c5659db00e.png)
![cineastError](https://user-images.githubusercontent.com/20126691/150570194-e055bbb0-a932-4263-a641-330faed4e3b8.png)

**Comment by lucaro** [comment, marked as answer]
Hey,

it looks like you might be using an incompatible combination of component versions. See https://github.com/vitrivr/cottontaildb/wiki/Compatibility-Matrix for an overview of what version of cottontail is compatible with what version of Cineast. Since cottontail is under heavy development, the most recent version might not yet be compatible with Cineast, and it looks like that might be the issue in your case.

**Reply by newtonsp** [reply]
Dear Luca,

Thank you for your quick answer. The error disappeared and I saved instances in the database correctly. 

For further reference, I'm using now cineast (master) and cottontaildb (0.12.13).

Sincerelly

**Reply by silvanheller** [reply]
Great, we should support cottontail master soon (https://github.com/vitrivr/cineast/pull/224)
